File: common/src/main/java/org/sandboxpowered/silica/network/handshake/clientbound/PingRequest.java
package org.sandboxpowered.silica.network.handshake.clientbound;

import org.sandboxpowered.silica.network.*;
import org.sandboxpowered.silica.network.handshake.serverbound.PongResponse;

public class PingRequest implements Packet {
    private long time;

    @Override
    public void read(PacketByteBuf buf) {
        this.time = buf.readLong();
    }

    @Override
    public void write(PacketByteBuf buf) {
        buf.writeLong(this.time);
    }

    @Override
    public void handle(PacketHandler packetHandler, Connection connection) {
        packetHandler.sendPacket(new PongResponse(time));
    }
}

File: common/src/main/kotlin/org/sandboxpowered/silica/network/handshake/clientbound/PingRequest.kt
package org.sandboxpowered.silica.network.handshake.clientbound

import org.sandboxpowered.silica.network.Connection
import org.sandboxpowered.silica.network.Packet
import org.sandboxpowered.silica.network.PacketByteBuf
import org.sandboxpowered.silica.network.PacketHandler
import org.sandboxpowered.silica.network.handshake.serverbound.PongResponse

class PingRequest(private var time: Long) : Packet {

    constructor() : this(-1)

    override fun read(buf: PacketByteBuf) {
        time = buf.readLong()
    }

    override fun write(buf: PacketByteBuf) {
        buf.writeLong(time)
    }

    override fun handle(packetHandler: PacketHandler, connection: Connection) {
        packetHandler.sendPacket(PongResponse(time))
    }
}